lib/core/test/unit/Cardano/Wallet/ApiSpec.hs

```diff
@@ -218,31 +218,37 @@ instance (GenericApiSpec a, GenericApiSpec b) => GenericApiSpec (a :<|> b) where
 instance GenericApiSpec Request where
     gSpec _ _ = pure ()
 
+instance GenericApiSpec a => GenericApiSpec [a] where
+    gSpec xs toSpec = foldr (\x y -> gSpec x toSpec >> y) (pure ()) xs
+
 instance
     ( Typeable a, Malformed (PathParam a)
-    ) => GenericApiSpec (PathParam a -> Request)
+    ) => GenericApiSpec (PathParam a -> [Request])
   where
-    gSpec toRequest toSpec = toSpec $
-        SomeTest (Proxy @a) (first toRequest <$> malformed @(PathParam a))
+    gSpec toRequest toSpec = do
+        let tests = fmap (\(xs, e) -> fmap (,e) xs)
+                (first toRequest <$> malformed @(PathParam a))
+        forM_ tests $ (toSpec . SomeTest (Proxy @a))
```

Comment:
Suggestion: Remove the extraneous `$`. (I think it's unnecessary here, given that you already have the parentheses.)

```suggestion
        forM_ tests (toSpec . SomeTest (Proxy @a))
```